src/ui/Distance/DistanceResources.ts

```diff
@@ -244,47 +246,42 @@ export class DistanceResources extends Component {
-    // Since this DistanceResource component often blocks initial queries,
-    // and relaunch the query automatically when it is able to do so (when a position provider resolves)
-    // we need to have a mechanism to still log something useful for usage analytics, instead of always a "position set".
-    // We want it to be "interface load" on a direct page access, or a "search from link" if coming from an external standalone search box
-    // Everytime this component blocks a query, we keep a copy of the pending search event, and resend this instead of a generic "position set" event.
-    if (this.pendingSearchEventOnCancellation) {
-      this.usageAnalytics.logSearchEvent(this.getPendingEventOnCancellation(), this.pendingSearchEventOnCancellation.getEventMeta());
-      delete this.pendingSearchEventOnCancellation;
-    } else {
-      this.usageAnalytics.logSearchEvent(analyticsActionCauseList.positionSet, {});
-    }
-    this.queryController.executeQuery();
+  private sendAnalytics() {
+    this.usageAnalytics.logSearchEvent(analyticsActionCauseList.positionSet, {});
   }
 
   private shouldTriggerQueryWhenPositionSet() {
-    // If query has been cancelled, we need to trigger the first one.
-    const triggerFirstQueryWithPosition = this.options.cancelQueryUntilPositionResolved && !this.isFirstPositionResolved;
-    return this.options.triggerNewQueryOnNewPosition || triggerFirstQueryWithPosition;
+    // Don't trigger the query if the first one is not yet executed.
+    return !this.queryController.firstQuery && this.options.triggerNewQueryOnNewPosition;
   }
 
-  private onAfterComponentsInitialization(): void {
+  private onAfterComponentsInitialization(afterComponentsInitializationArgs: IInitializationEventArgs): void {
     const args: IResolvingPositionEventArgs = {
       providers: this.getProvidersFromOptions()
     };
 
     this.bind.trigger(this.element, DistanceEvents.onResolvingPosition, args);
 
-    this.lastPositionRequest = this.tryGetPositionFromProviders(args.providers)
-      .then(position => {
-        if (position) {
-          this.setPosition(position.latitude, position.longitude);
-        } else {
-          this.triggerDistanceNotSet();
-        }
-        return position;
-      })
-      .catch(error => {
-        this.logger.error('An error occurred while trying to resolve the current position.', error);
+    this.lastPositionRequest = this.tryToSetPositionFromProviders(args.providers);
+
+    if (this.options.cancelQueryUntilPositionResolved) {
+      afterComponentsInitializationArgs.defer.push(this.lastPositionRequest);
+    }
+  }
+
+  private async tryToSetPositionFromProviders(providers: IGeolocationPositionProvider[]): Promise<IGeolocationPosition> {
+    try {
+      const position = await this.tryGetPositionFromProviders(providers);
+      if (position) {
+        this.setPosition(position.latitude, position.longitude);
+      } else {
         this.triggerDistanceNotSet();
```

Comment:
Edit: maybe the callee method show throw an error instead. I commented on the next file.

I would prefer to throw here and let the catch take care of triggering the event. As-is, we would not log an error message.

Could look something like:

```
private async tryToSetPositionFromProviders(providers: IGeolocationPositionProvider[]): Promise<IGeolocationPosition> {
    try {
        const position = await this.tryGetPositionFromProviders(providers);
        
        if (!position) {
            throw new Error('message');
        }
        
        this.setPosition(position.latitude, position.longitude);
    } catch (e) {
        // handle error
    }
}
```